From 98e6211d71ffae951ce4fc06309e8347420374a0 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@protonmail.com>
Date: Thu, 24 Apr 2025 19:33:51 -0600
Subject: [PATCH 1/1] Fix compatibility with base-compat-0.13

Many major packages still depend on base-compat-0.13, making v0.14
unsuitable for the time being.

This instead requires base >=4.16 (>=ghc-9.2)

Signed-off-by: hololeap <hololeap@protonmail.com>
--- a/Hledger/Web/Handler/RegisterR.hs
+++ b/Hledger/Web/Handler/RegisterR.hs
@@ -9,7 +9,7 @@
 
 module Hledger.Web.Handler.RegisterR where
 
-import qualified Data.List.NonEmpty.Compat as NonEmpty  -- from base-compat for ghc 8.10
+import qualified Data.List.NonEmpty as NonEmpty
 import Data.List (intersperse, nub, partition)
 import qualified Data.Text as T
 import Text.Hamlet (hamletFile)
--- a/hledger-web.cabal
+++ b/hledger-web.cabal
@@ -161,8 +161,8 @@ library
   build-depends:
       Decimal >=0.5.1
     , aeson >=1 && <2.3
-    , base >=4.14 && <4.22
-    , base-compat >=0.14.0
+    , base >=4.16 && <4.22
+    , base-compat >=0.13
     , base64
     , blaze-html
     , blaze-markup
@@ -227,8 +227,8 @@ executable hledger-web
   ghc-options: -Wall -Wredundant-constraints -Wincomplete-record-updates -Wincomplete-uni-patterns -with-rtsopts=-T
   cpp-options: -DVERSION="1.42.1"
   build-depends:
-      base >=4.14 && <4.22
-    , base-compat >=0.14.0
+      base >=4.16 && <4.22
+    , base-compat >=0.13
     , hledger-web
   default-language: Haskell2010
   if flag(threaded)
@@ -250,8 +250,8 @@ test-suite test
   ghc-options: -Wall -Wredundant-constraints -Wincomplete-record-updates -Wincomplete-uni-patterns
   cpp-options: -DVERSION="1.42.1"
   build-depends:
-      base >=4.14 && <4.22
-    , base-compat >=0.14.0
+      base >=4.16 && <4.22
+    , base-compat >=0.13
     , hledger-web
   default-language: Haskell2010
   if flag(threaded)
-- 
2.49.0

